docs/source/transforms.rst

```diff
@@ -1,219 +1,144 @@
+:mod:`torchaudio.transforms` module contains common audio processings and feature extractions. The following diagram shows the relationship between some of the transforms available.
 
-.. autoclass:: TimeStretch
 
-  .. automethod:: forward
+.. image:: https://download.pytorch.org/torchaudio/tutorial-assets/torchaudio_feature_extractions.png
 
-:hidden:`Fade`
---------------
+Transforms are implemented using :class:`torch.nn.Module`. Common ways to build a processing pipeline is to define custom Module class or chain Modules together using :class:`torch.nn.Sequential`, then moving it to target device and data type.
 
-.. autoclass:: Fade
+.. code::
 
-  .. automethod:: forward
+   # Define custom feature extraction pipeline.
+   #
+   # 1. Resample audio
+   # 2. Convert to power spectrogram
+   # 3. Apply augmentations
+   # 4. Convert to mel-scale
+   #
+   class MyPipeline(torch.nn.Module):
+       def __init__(
+           self,
+           input_freq=16000,
+           resample_freq=8000,
+           n_fft=1024,
+           n_mel=256,
+           stretch_factor=0.8,
+       ):
+           super().__init__()
+           self.resample = Resample(orig_freq=input_freq, new_freq=resample_freq)
 
-:hidden:`Vol`
--------------
-
-.. autoclass:: Vol
-
-  .. automethod:: forward
-
-:hidden:`Loudness`
-------------------
-
-.. autoclass:: Loudness
-
-  .. automethod:: forward
-
-:hidden:`Feature Extractions`
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-:hidden:`Spectrogram`
----------------------
-
-.. autoclass:: Spectrogram
-
-  .. automethod:: forward
-
-:hidden:`InverseSpectrogram`
-----------------------------
-
-.. autoclass:: InverseSpectrogram
-
-  .. automethod:: forward
-
-:hidden:`MelSpectrogram`
-------------------------
+           self.spec = Spectrogram(n_fft=n_fft, power=2)
 
-.. autoclass:: MelSpectrogram
+           self.spec_aug = torch.nn.Sequential(
+               TimeStretch(stretch_factor, fixed_rate=True),
+               FrequencyMasking(freq_mask_param=80),
+               TimeMasking(time_mask_param=80),
+           )
 
-  .. automethod:: forward
+           self.mel_scale = MelScale(
+               n_mels=n_mel, sample_rate=resample_freq, n_stft=n_fft // 2 + 1)
 
-:hidden:`GriffinLim`
---------------------
+       def forward(self, waveform: torch.Tensor) -> torch.Tensor:
+           # Resample the input
+           resampled = self.resample(waveform)
 
-.. autoclass:: GriffinLim
+           # Convert to power spectrogram
+           spec = self.spec(resampled)
 
-  .. automethod:: forward
+           # Apply SpecAugment
+           spec = self.spec_aug(spec)
 
-:hidden:`MFCC`
---------------
+           # Convert to mel-scale
+           mel = self.mel_scale(spec)
 
-.. autoclass:: MFCC
+           return mel
 
-  .. automethod:: forward
 
-:hidden:`LFCC`
---------------
+.. code::
 
-.. autoclass:: LFCC
+   # Instantiate a pipeline
+   pipeline = MyPipeline()
 
-  .. automethod:: forward
+   # Move the computation graph to CUDA
```

Comment:
could be good to move this to a dtype as well for demonstration

Reply:
done